File: solutions/fortran/bob/1/bob.f90
module bob
  implicit none
contains

  function hey(statement)
    character(100) :: hey
    character(len=*), intent(in) :: statement
    hey = "Whatever."
    if (scan(statement, "abcdefghijklmnopqrstuvxyz")==0 .and. scan(statement, "ABCDEFGHIJKLMNOPQRSTUVWXYZ")>0 .and. &
    statement(len(statement):len(statement))=="?") then
      hey="Calm down, I know what I'm doing!"
    else if (scan(adjustr(statement), "?")>0 .and. statement(len(adjustr(statement)):len(adjustr(statement)))==" ") then
      hey="Sure."  
    else if (statement(len(adjustr(statement)):len(adjustr(statement)))=="?") then
      hey="Sure."
    else if (scan(statement, "abcdefghijklmnopqrstuvxyz")==0 .and. scan(statement, "ABCDEFGHIJKLMNOPQRSTUVWXYZ")>0) then
      hey="Whoa, chill out!"
    else if (statement=="") then
      hey="Fine. Be that way!"
    end if 
  end function hey

end module bob

File: solutions/python/bob/1/bob.py
def response(hey_bob):
    hey_bob=' '.join(hey_bob.split())
    if hey_bob.isupper() and hey_bob.endswith('?'):
        answer="Calm down, I know what I'm doing!"
    elif hey_bob.isupper():
        answer="Whoa, chill out!"
    elif hey_bob.endswith('?'):
        answer='Sure.'
    elif hey_bob=='':
        answer='Fine. Be that way!'
    else:
        answer = 'Whatever.'
    return answer
